src/models/userModel.js

```diff
@@ -1,4 +1,90 @@
 const mongoose = require('mongoose');
+
+const bcrypt = require('bcryptjs');
+
+const NotificationSchema = new mongoose.Schema({
+    title: String,
+    message: String,
+    status: {
+        type: String,
+        enum: ['unread', 'read'],
+        default: 'unread',
+    },
+    timestamp: {
+        type: Date,
+        default: Date.now,
+    },
+});
+
+const AddressSchema = new mongoose.Schema({
+    street: String,
+    city: String,
+    state: String,
+    country: String,
+    zip: Number,
+});
+
+const LocationSchema = new mongoose.Schema({
```

Comment:
mongoose has a built in location functionalities (for example finding the closest point to another point on map and ...)
for that you need to use this schema (based on mongoose documentation) :
`const locationSchema = new mongoose.Schema({
  point: {
    type: {
      type: String,
      enum: ['Point'],
      required: true,
    },
    coordinates: {
      type: [Number], // two items, first longitude, and second latitude
      required: true,
    },
  },
  country: {
    type: String,
    required: true,
  },
  city: {
    type: String,
    required: true,
  },
});`

this is what I used for the location of the events. and we don't need a street or detailed address. because the only functionality of this document is to find close events in the same city/country